packages/contracts/development/version_checksums/log_bytecode_history.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+# This script helps us checks that we didn't change our audited code when refactoring unless we intended to.
+# It runs solc with the --no-cbor-metadata flag to prevent solc from appending ipfs metadata, which includes a hash of the source code.
+# It will run back through the history up to the specified number of commits.
+# It assumes they all use solc 0.8.20. You need that binary installed. Adjust the SOLC variable if it's installed but not called "solc-0.8.20"
+# It will output a file for each version with the commit, the hash of the compiled code and whether we put in a comment to claim we didn't change it from the previous commit (on the next line)
+# It should put you back at the head of the original branch once it's finished.
+
+if [[ "$1" -lt "1" ]]; then
+    echo "Usage: ./log_bytecode_history.sh <number of commits to go back>"
```

Comment:
I think the dirname was forgotten

Reply:
When calling solc you need to pass the `--no-cbor-metadata` flag, otherwise it will include a hash of the source code in the binary and superficial changes will produce a different hash. I also did it with optimization turned off, I don't know if that makes a difference but it removes a possible source of weirdness.

Also I just changed the script to catch the compilation failing properly, previously if it failed it would just hash an empty string which was confusing. One you run out of recent commits and get to the old code it should stop breaking because the source code specifies an older solc, that's expected.

Reply:
Do you want to have a quick go at running the latest one while telling solc not to optimize of include metadata? Don't spend more than a few minute son it - it it works then we're good, if it doesn't then I'll try it on a clean install in case I screwed something up and I'm relying on something I haven't check in or something like that.

Reply:
So, I can't use it without optimizer, since it then tells me that the contract are too big.
And I was not able to install solc here on my machine. That would defintively take more time.
I think solcjs doesn't need this flag --no-cbor-metadata. But probably it also does not include the metadata as solc does

Reply:
Contracts too big should just be a warning not an error, and I'm almost certain that solcjs also appends metadata, otherwise you would have to specify different versions to validate anything.

But anyhow here's my output with `log_bytecode_history.sh 40`

[COMMIT_LOG_RealityETH-3.0.sol.log](https://github.com/RealityETH/reality-eth-monorepo/files/14020907/COMMIT_LOG_RealityETH-3.0.sol.log)
[COMMIT_LOG_RealityETH_ERC20-3.0.sol.log](https://github.com/RealityETH/reality-eth-monorepo/files/14020908/COMMIT_LOG_RealityETH_ERC20-3.0.sol.log)

The last few are the expected compile failures because the old contracts expect older solc.